### PR TITLE
Digital Subscription Checkout - Thank You

### PR DESCRIPTION
**Pull request by Ap0c**
## Why are you doing this?

Adding the thank you page to the digital subscription checkout.

[**Trello Card**](https://trello.com/c/7rD9E7RR/1781-thank-you-page-excluding-dd-and-marketing-permissions)

cc @JustinPinner 

## Changes

- Added a new component for the `HeadingBlock`, which we're going to be using on multiple pages.
- Created a genericised `SubscriptionsThankYou` component that includes the hero image, heading block and dotcom link. We're likely to be re-using this pattern on multiple pages.
- Added hero SVGs for the thank you page.
- Created a page-specific `AppSection` component, with copy and links to the app stores.
- Added a component to handle displaying the checkout stages (`CheckoutStage`).
- Updated the subscriptions checkout to use the `SimpleHeader` and call the staging component.
- Created actions and a reducer for the checkout.

## Screenshots

Mobile | Tablet | Desktop
-|-|-
thankyou-mobile | thankyou-tablet | thankyou-desktop

### CODE REVIEW

File: assets/pages/digital-subscription-checkout/digitalSubscriptionCheckoutReducer.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+const initialState = {
+  stage: 'checkout',
+};
+
+function reducer(state: PageState = initialState, action: Action): PageState {
+
+  switch (action.type) {
+
+    case 'SET_STAGE':
```
Comment (svillafe) [inline assets/pages/digital-subscription-checkout/digitalSubscriptionCheckoutReducer.js]: Is it possible to add tests for this reducer and action?

File: assets/components/svgs/subsThankYouHeroMobile.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  return (
+    <svg
+      className="svg-subs-thank-you-hero-mobile"
+      xmlns="http://www.w3.org/2000/svg"
+      viewBox="0 90 463 363"
+      aria-labelledby="svg-subs-thank-you-hero-mobile-title"
+    >
+      <title id="svg-subs-thank-you-hero-mobile-title">Thank You</title>
+      <g fill="none" fillRule="evenodd" transform="translate(0 -31)">
+        <circle cx="38.5" cy="231.5" fill="#FFE500" r="22.5" transform="matrix(-1 0 0 1 77 0)" />
```
Comment (svillafe) [inline assets/components/svgs/subsThankYouHeroMobile.jsx]: 👍 for using `<circle/>`

File: assets/components/subscriptionsThankYou/subscriptionsThankYou.jsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      <div className="component-subscriptions-thank-you__hero">
+        <SvgSubsThankYouHeroMobile />
+        <SvgSubsThankYouHeroTablet />
+        <SvgSubsThankYouHeroDesktop />
+      </div>
+      <LeftMarginSection>
+        <HeadingBlock heading="Thank you for helping to support our journalism">
+          <p className="component-subscriptions-thank-you__heading-banner">
+            <strong className="component-subscriptions-thank-you__congrats">
+              Congratulations
```
Comment (JustinPinner) [inline assets/components/subscriptionsThankYou/subscriptionsThankYou.jsx]: LOL

File: assets/pages/digital-subscription-checkout/digitalSubscriptionCheckoutReducer.js (hunk 1 above)
Comment (Ap0c) [reply assets/pages/digital-subscription-checkout/digitalSubscriptionCheckoutReducer.js]: Done.

**Pull request merged**